ga4gh/datamodel/rna_quantification.py

```diff
@@ -0,0 +1,392 @@
+        """
+        self._annotationIds = fields["annotation_ids"].split(',')
+        self._description = fields["description"]
+        self._name = fields["name"]
+        self._readGroupId = fields["read_group_id"]
+
+
+class RNASeqResult(AbstractRNAQuantification):
+    """
+    Class representing a single RnaQuantification in the GA4GH data model.
+    """
+
+    def __init__(self, parentContainer, localId, rnaQuantDataPath,
+                 dataRepository):
+        super(RNASeqResult, self).__init__(parentContainer, localId)
+        self._dbFilePath = rnaQuantDataPath  # the full path of the db file
+        self._dataRepository = dataRepository
+        self._db = SqliteRNABackend(self._dbFilePath)
+        self.getRnaQuantMetadata()
+
+    def getRnaQuantMetadata(self):
+        """
+        input is tab file with no header.  Columns are:
+        Id, annotations, description, name, readGroupId
+        where annotation is a comma separated list
+        """
+        rnaQuantId = self.getLocalId()
+        with self._db as dataSource:
+            rnaQuantReturned = dataSource.getRnaQuantificationById(
+                rnaQuantId)
+        self.addRnaQuantMetadata(rnaQuantReturned)
+
+    def getExpressionLevels(
+            self, rnaQuantID, pageToken=0, pageSize=None, expressionId=None,
+            featureGroupId=None, threshold=0.0):
+        """
+        Returns the list of ExpressionLevels in this RNA Quantification.
+        """
+        if expressionId is not None:
+            rawId = datamodel.ExpressionLevelCompoundId.deobfuscate(
+                expressionId)
+            expressionId = string.split(rawId, ":")[-1]
+        with self._db as dataSource:
+            expressionsReturned = dataSource.searchExpressionLevelsInDb(
+                rnaQuantID, pageToken=pageToken,
+                pageSize=pageSize, expressionId=expressionId,
+                featureGroupId=featureGroupId, threshold=threshold)
+        # TOOD: try to use a generator - something like below
+        # for expressionEntry in expressionsReturned:
+        #    yield ExpressionLevel(self, expressionEntry)
+        return [ExpressionLevel(self, expressionEntry) for
+                expressionEntry in expressionsReturned]
+
+    def getExpressionLevel(self, compoundId):
+        rawId = datamodel.ExpressionLevelCompoundId.deobfuscate(compoundId)
+        localExpressionId = string.split(rawId, ":")[-1]
+        with self._db as dataSource:
+            expressionReturned = dataSource.getExpressionLevelById(
+                localExpressionId)
+
+        return ExpressionLevel(self, expressionReturned)
+
+    def getFeatureGroups(
+            self, rnaQuantID, pageToken=0, pageSize=None, featureGroupId=None):
+        """
+        Returns the list of FeatureGroups in this RNA Quantification.
+
+        for now the feature group data is autogenerated by examining the
+        relevant expression data file
```

Comment:
Ah, I see! Although it's confusing to me that a `FeatureGroup` is effectively an ephemeral entity computed in the RNA code, while `FeatureSet` is a straightforward container entity from Sequence Annotation. Is there a more descriptive name for this?